CMakeLists.txt

```diff
@@ -4,33 +4,80 @@ project(ccpp_framework
         VERSION 5.0.0
         LANGUAGES Fortran)
 
+enable_language(C CXX)
+include(FetchContent)
+include(cmake/ccpp_capgen.cmake)
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${PROJECT_SOURCE_DIR}/cmake")
+
+
 #------------------------------------------------------------------------------
 # Set package definitions
 set(PACKAGE "ccpp-framework")
 set(AUTHORS "Dom Heinzeller" "Grant Firl" "Mike Kavulich" "Dustin Swales" "Courtney Peverley")
 string(TIMESTAMP YEAR "%Y")
 
+option(CCPP_FRAMEWORK_BUILD_DOCUMENTATION
+       "Create and install the HTML documentation (requires Doxygen)" OFF)
+option(CCPP_FRAMEWORK_ENABLE_OPENMP "Enable OpenMP support for the framework" OFF)
+option(CCPP_FRAMEWORK_ENABLE_TESTS "Enable building/running tests" OFF)
+option(BUILD_SHARED_LIBS "Build a static library" OFF)
+set(CCPP_VERBOSITY "0" CACHE STRING "Verbosity level of output (default: 0)")
+
+if(CCPP_FRAMEWORK_ENABLE_TESTS)
```

Comment:
If `CCPP_FRAMEWORK_ENABLE_TESTS` is ON, then there needs to be a `find_package` call for `pFUnit` with the `REQUIRED` keyword. pfunit must come from the user-provided software stack. No implicit installs during cmake calls/build steps.

Reply:
Understood, happy to make that change.  I'm actually looking to make a container with all the dependencies/packages already installed so the environment better mimics our development environments to make the cmake call clean as well (we can pass in the pfunit path at configure time but it's clunky without modules).